juju/juju
# k8s brokers implement SetCloudSpec, respond to update-cloud

## Description of change

The k8s broker implements SetCloudSpec(). This allows it to respond to cloud changes. The fix needed to separate out the actual k8s API client so it could be set via a mutex after initial creation.

Additionally, the broker tracker worker has a bunch of code copied across from the environ tracker worker. The environ tracker already implemented the necessary watchers etc for vm clouds. 

## QA steps

bootstrap k8s
run update-cloud with a changed cloud definition
see that the logs show the tracker worker responding to the change
